### PR TITLE
Fix SDK integration tests in local dev and add CI step to verify

### LINKED ISSUE #1597: The `standalone-integration-tests-check` CI job went missing
The `standalone-integration-tests-check` CI job went missing, and now the SDK's manual integration test `Cargo.toml` files no longer match the generated `Cargo.toml` files, which makes local development much harder since many of them don't even compile.

- [x] Restore checks for the manual `Cargo.toml` files (originally added in https://github.com/awslabs/smithy-rs/commit/21df7234a64a5f48d3b800158048dcf878295083)
- [x] Fix integration test builds

### PR DESCRIPTION
**Pull request by jdisanti**
## Motivation and Context
Fixes #1597

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws/rust-runtime/aws-inlineable/Cargo.toml [hunk 1]
```diff
@@ -29,7 +29,7 @@ http-body = "0.4.5"
 md-5 = "0.10.1"
 ring = "0.16"
 tokio = { version = "1", features = ["full"] }
-tokio-stream = "0.1.7"
+tokio-stream = "0.1.5"
```
Comment (rcoh) [inline aws/rust-runtime/aws-inlineable/Cargo.toml:32]: intentional downgrade?

File: aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/IntegrationTestDependencies.kt [hunk 2]
```diff
@@ -110,14 +110,14 @@ class S3TestDependencies(
...
+private val AsyncStream = CargoDependency("async-stream", CratesIo("0.3.0"), DependencyScope.Dev)
+private val Criterion = CargoDependency("criterion", CratesIo("0.3.0"), scope = DependencyScope.Dev)
+private val FuturesCore = CargoDependency("futures-core", CratesIo("0.3.0"), DependencyScope.Dev)
+private val FuturesUtil = CargoDependency("futures-util", CratesIo("0.3.0"), scope = DependencyScope.Dev)
+private val Hound = CargoDependency("hound", CratesIo("3.4.0"), DependencyScope.Dev)
+private val SerdeJson = CargoDependency("serde_json", CratesIo("1.0.0"), features = emptySet(), scope = DependencyScope.Dev)
+private val Smol = CargoDependency("smol", CratesIo("1.2.0"), scope = DependencyScope.Dev)
+private val Tokio = CargoDependency("tokio", CratesIo("1.0.0"), features = setOf("macros", "test-util"), scope = DependencyScope.Dev)
+private val Tracing = CargoDependency("tracing", CratesIo("0.1.0"), scope = DependencyScope.Dev)
+private val TracingSubscriber = CargoDependency("tracing-subscriber", CratesIo("0.3.5"), scope = DependencyScope.Dev, features = setOf("env-filter"))
```
Comment (rcoh) [inline aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/IntegrationTestDependencies.kt:123]: we should build a dependency version system

**Review by rcoh — APPROVED** [review]
LGTM—we don't need `1.0.0` because CI will fail if the requested version doesn't contain all the required APIs

File: aws/rust-runtime/aws-inlineable/Cargo.toml (hunk 1 above)
Comment (jdisanti) [reply aws/rust-runtime/aws-inlineable/Cargo.toml:32]: Yeah, I picked the minimum version across all the `tokio-stream` references.

File: aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/IntegrationTestDependencies.kt (hunk 2 above)
Comment (jdisanti) [reply aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/IntegrationTestDependencies.kt:123]: Tracking in https://github.com/awslabs/smithy-rs/issues/1611

**Pull request merged**